dns/google/cloud/dns/zone.py

```diff
@@ -373,26 +365,46 @@ def list_changes(self, max_results=None, page_token=None, client=None):
         :param client: the client to use.  If not passed, falls back to the
                        ``client`` stored on the current zone.
 
-        :rtype: tuple, (list, str)
-        :returns: list of
-                  :class:`~.resource_record_set.ResourceRecordSet`,
-                  plus a "next page token" string:  if the token is not None,
-                  indicates that more zones can be retrieved with another
-                  call (pass that value as ``page_token``).
+        :rtype: :class:`~google.cloud.iterator.Iterator`
+        :returns: Iterator of :class:`~.changes.Changes`
+                  belonging to this zone.
         """
-        params = {}
+        client = self._require_client(client)
+        path = '/projects/%s/managedZones/%s/changes' % (
+            self.project, self.name)
+        iterator = Iterator(
+            client=client, path=path, items_key='changes',
+            item_to_value=_item_to_changes,
+            page_token=page_token, max_results=max_results)
+        iterator.zone = self
```

Comment:
I was initially puzzled by this attribute, but see that it is needed for `_item_to_changes`, which makes sense.